src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2835,6 +2838,11 @@ void processContextElement
       responseP->contextElementResponseVector.push_back(cerP);
     }
   }
+  else if (apiVersion == "v2" && action == "APPEND" && !isUpdate)
```

Comment:
Looking to this else in context:

```
 if (docs == 0)
 {
 ...
 }
 else if (apiVersion == "v2" && action == "APPEND" && !isUpdate)
 {
 ...
 }
```

Note that the `if(doc == 0)` block is processed after the `while (cursor->more())` block in which the result of the query that get the entity matching with the update is processed. Thus, as far as I understand, you capture the error *after* the existing entity has been modified in DB, which wouldn't be a right solution as modifications in the existing entity have to be avoided.

If my theory is correct, we should see the fail when executing the the .test case you are now implementing to cover this fix. Thus I'd recomment first you develop the .test and once we check with that .test that current way doesn't work, look for a better solution (probably similar to the one you proposed in a previous attemp https://github.com/telefonicaid/fiware-orion/pull/1193#discussion_r38410801, based in detecting duplicated entities before the `while (cursor->more())` block.